session/session.go

```diff
@@ -18,20 +18,20 @@ func generateSecret(base, salt string) []byte {
 	return pbkdf2.Key([]byte(base), []byte(salt), keyIterNum, keySize, sha1.New)
 }
 
-// I stolen the snippet from: https://gist.github.com/doitian/2a89dc9e4372e55335c9111f576b47bf
-func verifySign(encryptedData, sign, base, signSalt string) bool {
+// The origin of this snippet can be found at https://gist.github.com/doitian/2a89dc9e4372e55335c9111f576b47bf
+func verifySign(encryptedData, sign, base, signSalt string) (bool, error) {
 	signKey := generateSecret(base, signSalt)
 	signHmac := hmac.New(sha1.New, signKey)
 	signHmac.Write([]byte(encryptedData))
 	verifySign := signHmac.Sum(nil)
 	signDecoded, err := hex.DecodeString(sign)
 	if err != nil {
-		return false
+		return false, err
 	}
 	if hmac.Equal(verifySign, signDecoded) {
-		return true
+		return true, nil
 	}
-	return false
+	return false, errors.New("session: signature not verified")
```

Comment:
You might also consider using a guard-clause style, i.e.
```go
if !hmac.Equal(/* ... */) {
    return false, ErrInvalidSignature
}

return true, nil
```

Reply:
yes, you're right.